packages/services/api/src/modules/organization/providers/organization-access-tokens.ts

```diff
@@ -315,31 +567,533 @@ export class OrganizationAccessTokens {
+
+    return OrganizationAccessTokensPermissions.permissionGroups
+      .map(group => ({
+        ...group,
+        permissions: group.permissions
+          .filter(permission => filter(permission.id))
+          .map(permission => ({
+            ...permission,
+            isAssignableByViewer: true,
+          })),
+      }))
+      .filter(group => group.permissions.length !== 0);
+  }
+
+  async getAvailablePermissionGroupsForMembership(membership: OrganizationMembership) {
+    const organization = await this.storage.getOrganization({
+      organizationId: membership.organizationId,
+    });
+    const groups = await this.getAvailablePermissionGroupsForOrganization(organization);
+    const memberPermissions = membership.assignedRole.role.allPermissions;
+
+    return groups
+      .map(group => ({
+        ...group,
+        permissions: group.permissions.map(permission => ({
+          ...permission,
+          isAssignableByViewer: memberPermissions.has(permission.id),
+        })),
+      }))
+      .filter(group => group.permissions.length !== 0);
+  }
+
+  async getPermissionsForAccessToken(accessToken: OrganizationAccessToken) {
+    if (!accessToken.userId) {
+      return accessToken.permissions ?? [];
+    }
+    const organization = await this.storage.getOrganization({
+      organizationId: accessToken.organizationId,
+    });
+    const membership = await this.members.findOrganizationMembership({
+      organization,
+      userId: accessToken.userId,
+    });
+    if (!membership) {
+      return [];
+    }
+
+    const allRolePermissions = membership.assignedRole.role.allPermissions;
+
+    if (!accessToken.permissions) {
+      return Array.from(membership.assignedRole.role.allPermissions);
+    }
+
+    return accessToken.permissions.filter(permission => allRolePermissions.has(permission));
+  }
+
+  async getResourceAssignmentsForAccessToken(accessToken: OrganizationAccessToken) {
+    if (accessToken.userId === null) {
+      return this.resourceAssignments.resolveGraphQLMemberResourceAssignment({
+        organizationId: accessToken.organizationId,
+        resources: accessToken.assignedResources,
+      });
+    }
+
+    const organization = await this.storage.getOrganization({
+      organizationId: accessToken.organizationId,
+    });
+    const membership = await this.members.findOrganizationMembership({
+      organization,
+      userId: accessToken.userId,
+    });
+
+    // TODO: we could handle this different TBH
+    if (!membership) {
+      return {
+        mode: 'GRANULAR',
+      } satisfies GraphQLSchema.ResolversTypes['ResourceAssignment'];
+    }
+
+    return this.resourceAssignments.resolveGraphQLMemberResourceAssignment({
+      organizationId: accessToken.organizationId,
+      resources: intersectResourceAssignments(
+        accessToken.assignedResources,
+        membership?.assignedRole.resources,
+      ),
+    });
+  }
+
+  getGraphQLResolvedResourcePermissionGroupForAccessToken =
```

Comment:
This one is a bit tricky... 😅 

This "resolves" all the permissions and resources from a access token (or access token like for compatibility with legacy access tokens), and exposes them in a easy to consume format for both the CLI and Web App Access Token Detail View.

It is not perfect, as it could give more insights for the following:
- PATs: show which permissions where granted, but got revoked because of membership permission changes (same for resources)
- deleted resources: show deleted resources, that no longer exists using their UUID instead

However, the latter is something for a follow up pull request to reduce the already high complexity in this PR.